subsys/modbus/modbus_server.c

```diff
@@ -995,10 +1016,24 @@ bool modbus_server_handler(struct modbus_context *ctx)
 		break;
 
 	default:
-		LOG_ERR("Function code 0x%02x not implemented", fc);
-		mbs_exception_rsp(ctx, MODBUS_EXC_ILLEGAL_FC);
-		send_reply = true;
-		break;
+		if (IS_ENABLED(CONFIG_MODBUS_USER_DEFINED_FC)) {
```

Comment:
We should just call `send_reply = mbs_try_user_fc(...);` and handle user callback as well mbs_exception_rsp() there.